### PR TITLE
use command flag

### LINKED ISSUE #2216: Use the command flag to replace arguments for ebrelayer
Command flag is better way to input parameters for ebrelayer, it is more flexible.

### PR DESCRIPTION
**Pull request by juniuszhou**
The pr remove the arguments in the command line for ebrelayer, instead use the flag. will resolves #2216

### CODE REVIEW

File: cmd/ebrelayer/main.go [hunk 1]
```diff
@@ -165,35 +167,40 @@ func RunInitRelayerCmd(cmd *cobra.Command, args []string) error {
 	}
 
 	// Validate and parse arguments
-	networkDescriptor, err := strconv.Atoi(args[0])
+	networkDescriptor, err := cmd.Flags().GetInt32(networkDescriptorFlag)
 	if err != nil {
-		return errors.Errorf("%s is invalid network id", args[0])
+		return err
```
Comment (Brando753) [inline cmd/ebrelayer/main.go:172]: Is the new error message descriptive or generic? Does it specify that the network id flag was missing/invalid?

File: cmd/ebrelayer/main.go [hunk 2]
```diff
@@ -370,6 +381,26 @@ func AddRelayerFlagsToCmd(cmd *cobra.Command) {
 		int32(oracleTypes.NetworkDescriptor_NETWORK_DESCRIPTOR_ETHEREUM),
 		"The network descriptor for the chain",
 	)
+	cmd.Flags().String(
```
Comment (Brando753) [inline cmd/ebrelayer/main.go:384]: Are these flags being marked required or does the fetch and check error handle that automatically?

File: smart-contracts/scripts/devenv.ts [hunk 3]
```diff
@@ -118,7 +118,9 @@ async function main() {
       sifnode.results.relayerAddresses[0],
       sifnode.results.witnessAddresses[0],
       golang.results,
-      hardhat.results.chainId
```
Comment (Brando753) [inline smart-contracts/scripts/devenv.ts:121]: Is there a reason this value is not able to come from the hardhat builder?

**Review by Brando753 — COMMENTED** [review]
Once the questions are answered I will approve.

File: smart-contracts/scripts/devenv.ts (hunk 3 above)
Comment (juniuszhou) [reply smart-contracts/scripts/devenv.ts:121]: The chainId from hardhat is 1, we need to configure it as 31337.

File: cmd/ebrelayer/main.go (hunk 2 above)
Comment (juniuszhou) [reply cmd/ebrelayer/main.go:384]: I didn't find any parameter to set the flag as mandatory, and cobra will check the validatity.

File: cmd/ebrelayer/main.go (hunk 1 above)
Comment (juniuszhou) [reply cmd/ebrelayer/main.go:172]: it is a generic error description.

Comment (Brando753) [reply cmd/ebrelayer/main.go:172]: Can we make this specific? The problem when using this CLI is when a flag is incorrect/missing generic error messages make it a real pain to debug. What happens if the flag is missing here does it return an error?

Comment (juniuszhou) [reply cmd/ebrelayer/main.go:172]: OK, will add specific error info in the command line.

**Review by Brando753 — APPROVED** [review]

**Pull request merged**